libavformat/hls.c

```diff
@@ -227,11 +227,11 @@ typedef struct HLSContext {
 } HLSContext;
 
 static int is_variant_selected(HLSContext* c, const char* current_bandwidth) {
-    if (!c->selected_bandwidth && c->selected_variant_index == -1) {
+    if (!*c->selected_bandwidth && c->selected_variant_index == -1) {
```

Comment:
I think we need an if check at the top to return early if c->selected_bandwidth is NULL or current_bandwidth is NULL.  Otherwise we could dereference a NULL pointer later on.

Reply:
Unless we are always expecting a value (eg. "\0" if not set)